### DISCUSSION TITLE
error LNK2019 when use *.hpp by xrepo

### DISCUSSION THREAD

**Discussion by imLeonardo** [xmake-io/xmake#5321] [Q&A] [answered]
使用xrepo安装了lexy，但是在编译时提示找不到符号
xrepo install lexy

xmake.lua:
...
add_requires("lexy")
...
target("xxx", function()
  ...
    add_packages("lexy")
  ...
end)

编译时报 error LNK2019: 无法解析的外部符号 "bool __cdecl lexy::_detail::code_point_has_properties<0>(char32_t)"，请问怎么配置编译xrepo安装的库中的*.hpp文件

**Comment by waruqi** [comment]
如果是 headeronly 库，xmake 只要能保证 include 正常就行了，其他就跟 xmake 没啥关系了，你得去问库作者。。除非它不是 headeronly 的，还需要编译生成 .o，那就提 pr 改成 static/shared 库。

**Comment by imLeonardo** [comment]
很奇怪，我两台电脑，一样的代码，一样的xmake版本，一台报这个错，另一个正常编译

**Comment by imLeonardo** [comment]
应该是本地环境问题，我在查下

**Reply by waruqi** [reply]
可以看下编译器版本差异

**Reply by imLeonardo** [reply]
好的，感谢

**Reply by imLeonardo** [reply, marked as answer]
找到问题了，是有个函数库里面只有定义没有实现，需要自己实现
